mammoth/modules/adapters.py

```diff
@@ -201,7 +202,7 @@ def _inject_adapters(self):
         ):
             if layer_type == 'f':
                 # Adapters apply to feedforward layers
-                adapter_layers = adapter_layers_by_index[i]
+                adapter_layers = adapter_layers_by_index[f'layer{i}']
```

Comment:
not sold, but okay

Reply:
Yeah, not the best of solutions.

The reasoning is that `adapter_layers` is gappy (not all indices need to be populated), and is therefore a `ModuleDict` instead of a `ModuleList` (although the latter with placeholder `None`s would also work, with slightly more complicated code). The dot notation used for module hierarchies doesn't separate dicts and lists, except that the latter has to use integer indices. Using integer keys in a dict could potentially cause some confusion. Not sure if worth it though.